pkgs/applications/virtualization/docker/default.nix

```diff
@@ -1,78 +1,156 @@
-{ stdenv, lib, fetchFromGitHub, makeWrapper, pkgconfig, go-md2man
-, go, containerd, runc
-, sqlite, iproute, bridge-utils, devicemapper, systemd
+{ stdenv, lib, fetchFromGitHub, makeWrapper, cmake, pkgconfig
+, go-md2man, libapparmor, apparmor-parser, libseccomp, git
+, go, sqlite, iproute, bridge-utils, devicemapper, systemd
 , btrfs-progs, iptables, e2fsprogs, xz, utillinux, xfsprogs
 , procps
 }:
 
 # https://github.com/docker/docker/blob/master/project/PACKAGERS.md
-
 with lib;
-
-stdenv.mkDerivation rec {
+let
   name = "docker-${version}";
-  version = "1.12.3";
-
-  src = fetchFromGitHub {
+  version = "1.13.0-rc2";
+  dockerSrc = fetchFromGitHub {
     owner = "docker";
     repo = "docker";
-    rev = "v${version}";
-    sha256 = "0jifd35h22lgh36w1j2k97pgndjh5sppr3cwndlv0saf9618wx5k";
+    rev = "1f9b3ef9e9c74c2905780ac5347d018646a20022";
+    sha256= "1z44k5csnjxqrsaxvc6faych8dyn4y150rgcwk0b19nqiy9ld0rg";
+  };
+  libnetworkSrc = fetchFromGitHub {
+    owner = "docker";
+    repo = "libnetwork";
+    rev = "0f534354b813003a754606689722fe253101bc4e";
+    sha256 = "1ah7h417llcq0xzdbp497pchb9m9qvjhrwajcjb0ybrs8v889m31";
+  };
+  runcSrc = fetchFromGitHub {
+    owner = "docker";
+    repo = "runc";
+    rev = "51371867a01c467f08af739783b8beafc154c4d7";
+    sha256 = "1vyc6c59npwv8xyc1qynzz5i7kwnjgf5yda4avfnap8z4zn14m0x";
+  };
+  containerdSrc = fetchFromGitHub {
+    owner = "docker";
+    repo = "containerd";
+    rev = "03e5862ec0d8d3b3f750e19fca3ee367e13c090e";
+    sha256 = "184sd9dwkcba3zhxnz9grw8p81x05977p36cif2dgkhjdhv12map";
+  };
+  tiniSrc = fetchFromGitHub {
+    owner = "krallin";
+    repo = "tini";
+    rev = "949e6facb77383876aeff8a6944dde66b3089574";
+    sha256 = "0zj4kdis1vvc6dwn4gplqna0bs7v6d1y2zc8v80s3zi018inhznw";
   };
+in
+stdenv.mkDerivation {
+  inherit name version;
+
+  srcs = [ dockerSrc libnetworkSrc runcSrc containerdSrc tiniSrc ];
+  sourceRoot = ".";
+
+  hardeningDisable = ["fortify"];
 
   buildInputs = [
-    makeWrapper pkgconfig go-md2man go
+    makeWrapper cmake pkgconfig go-md2man go git
     sqlite devicemapper btrfs-progs systemd
+    libapparmor apparmor-parser libseccomp
   ];
 
   dontStrip = true;
 
+  dontUseCmakeConfigure = true;
+
   DOCKER_BUILDTAGS = []
     ++ optional (systemd != null) [ "journald" ]
     ++ optional (btrfs-progs == null) "exclude_graphdriver_btrfs"
     ++ optional (devicemapper == null) "exclude_graphdriver_devicemapper";
 
+    postUnpack = ''
+    cp -fR ${dockerSrc} docker
```

Comment:
I did get errors with chmod… I'm not used to multiple sources build so I *hacked* around for that… There might be a better solution 👼